### PR TITLE
Add JTAG tutorial

### PR DESCRIPTION
**Pull request by andre-richter**
@naotaco 

Could you please review? This branch contains your changes and my extensions.

You can check out the `JTAG` branch of the repository for this.

### CODE REVIEW

**Comment by naotaco** [comment]
@andre-richter 

Wow, it's really nice tutorial. Great job! 👏  Honestly, I had no Idea to provide simple way to load another image to debug after enable/connect JTAG debugger.

Anyway, I'll follow the tutorial. Give me few days for that. At least, I'll provide improved wiring photo and fritzing image as requested at #18 .

File: 0B_hw_debug_JTAG/src/power.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            mbox.buffer[6] = 0; // bit 0: off, bit 1: no wait
+            mbox.buffer[7] = mbox::tag::LAST;
+
+            // Insert a compiler fence that ensures that all stores to the
+            // mbox buffer are finished before the GPU is signaled (which
+            // is done by a store operation as well).
+            compiler_fence(Ordering::Release);
+
+            if mbox.call(mbox::channel::PROP).is_err() {
+                return Err(PowerError::MailboxError); // Abort if UART clocks couldn't be set
```
Comment (berkus) [inline 0B_hw_debug_JTAG/src/power.rs]: Invalid comment - we turn off POWER, not set UART clocks.

**Comment by berkus** [comment]
Thank you @andre-richter for this; my journey with JTAG debugging my Pi was bumpy up to this point, hopefully, it will become a smoother ride now!

I will play with the 0E tutorial and write up if anything is not working. I have a segger JLink not olimex, so things might be slightly different.

File: 0B_hw_debug_JTAG/src/power.rs (hunk 1 above)
Comment (andre-richter) [reply 0B_hw_debug_JTAG/src/power.rs]: Good catch, thanks. Corrected it.

**Comment by andre-richter** [comment]
@berkus Using `config.txt` to set the GPIOs is something I was ignorant about, it can definitely be an alternative. Thanks for sharing.

For the tutorials, since we have our chain-loading infrastructure with `raspboot` in place anyways, I'll stick with the current solution. It has the advantage that the pins are only set on demand when needed, and not on every boot regardless.

**Comment by andre-richter** [comment]
@naotaco @berkus
I think my exit criteria for merging this one to master is that one of you guys could run and step with gdb successfully.

All points raised in #18 and other findings can be addressed in master then.

**Comment by naotaco** [comment]
@andre-richter 

I agree. Actually, gdb is working here perfectly! I can break the CPU and see registers/source code in cool gdb window. Thanks again for your great work ❤️

**Comment by andre-richter** [comment]
Perfect, thanks.

**Pull request merged**